convert2rhel/logger.py

```diff
@@ -299,33 +268,63 @@ def disable_colors(self, value):
         self.color_disabled = value
 
     def format(self, record):
-        if record.levelno == LogLevelTask.level:
-            temp = "*" * (90 - len(record.msg) - 25)
-            fmt_orig = "\n[%(asctime)s] %(levelname)s - [%(message)s] " + temp
-            new_fmt = fmt_orig if self.color_disabled else colorize(fmt_orig, "OKGREEN")
-            self._fmt = new_fmt
+        """Format tasks, etc
+
+        :param logging.LogRecord record: Logger-provided LogRecord that is
+        provided when we use logging.warning() etc.
+        :return str: The formatted log message
+        """
+        fmt_orig = "[%(asctime)s] %(levelname)s - %(message)s"  # DEBUG default
+        self.datefmt = "%Y-%m-%dT%H:%M:%S%z"  # DEBUG default
+
+        # Used when providing isTask in the extra field
+        # e.g. logging.warning("Testing", extra={"isTask": True})
+        isTask = getattr(record, "isTask", False)
+
+        color = self._getLogLevelColor(record, isTask)
+        if isTask:
+            asterisks = "*" * (90 - len(record.msg) - 25)
+            fmt_orig = "\n[%(asctime)s] TASK - [%(message)s] " + asterisks
             self.datefmt = "%Y-%m-%dT%H:%M:%S%z"
-        elif record.levelno in [logging.INFO]:
-            self._fmt = "%(message)s"
-            self.datefmt = ""
-        elif record.levelno in [logging.WARNING]:
-            fmt_orig = "%(levelname)s - %(message)s"
-            new_fmt = fmt_orig if self.color_disabled else colorize(fmt_orig, "WARNING")
-            self._fmt = new_fmt
-            self.datefmt = ""
-        elif record.levelno >= logging.ERROR:
-            # Error, Critical, Critical_no_exit
+        elif record.levelno >= logging.WARNING:
             fmt_orig = "%(levelname)s - %(message)s"
-            new_fmt = fmt_orig if self.color_disabled else colorize(fmt_orig, "FAIL")
-            self._fmt = new_fmt
+        elif record.levelno >= logging.INFO:
+            fmt_orig = "%(message)s"
             self.datefmt = ""
-        else:
-            self._fmt = "[%(asctime)s] %(levelname)s - %(message)s"
-            self.datefmt = "%Y-%m-%dT%H:%M:%S%z"
 
+        new_fmt = fmt_orig if not color or self.color_disabled else colorize(fmt_orig, color)
+        self._fmt = new_fmt
         if hasattr(self, "_style"):
             # Python 3 has _style for formatter
             # Overwriting the style _fmt gets the result we want
             self._style._fmt = self._fmt
 
         return super(CustomFormatter, self).format(record)
+
+    def _getLogLevelColor(self, record, isTask=False):
+        if isTask:
+            return "OKGREEN"
+        elif record.levelno >= logging.WARNING:
+            return "WARNING"
+        elif record.levelno >= logging.ERROR:
+            return "FAIL"
+        return None
+
+
+class CustomLogger(logging.Logger):
+    def __init__(self, name, level=0):
```

Comment:
Can we have a docstring here mentioning why this class exist? Looking at it and understanding the context of the PR is easy to guess, but future people might not get it at first.

Reply:
Especially if you can include that the `critical_no_exit`, `task` and `file` were moved here to avoid messing up with the parent logger class